File: Java/2951-3000/2962. Count Subarrays Where Max Element Appears at Least K Times.java
// https://leetcode.com/problems/count-subarrays-where-max-element-appears-at-least-k-times/

public long countSubarrays(int[] nums, int k) {
        // Get the num with maximum count
        int maxNum = Arrays.stream(nums).max().getAsInt();
        int left = 0, right = 0, size = nums.length;

        long result = 0;

        while(right < size){
            k -= (nums[right++] == maxNum) ? 1 : 0;

            while(k == 0)
                k += (nums[left++] == maxNum) ? 1 : 0;

            result += left;
        }

        return result;
    }

File: Python/2951-3000/2962. Count Subarrays Where Max Element Appears at Least K Times.py
# https://leetcode.com/problems/count-subarrays-where-max-element-appears-at-least-k-times/

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        maxNum = max(nums) # Get the num with maximum count
        result = left = right = 0

        while right < len(nums):
            k -= 1 if (nums[right] == maxNum) else 0
            right += 1

            while k == 0:
                k += 1 if (nums[left] == maxNum) else 0
                left += 1

            result += left

        return result
